From d2f507fe050c5e1d6aee78384c37da816d2c2323 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Wed, 23 Sep 2020 17:33:42 -0400
Subject: [PATCH 1/2] filter: include: Mark move constructor/assignment as
 noexcept.

With this, MSVC will actually use the move operations instead of trying
to use the copy operations (and failing because they are deleted).
---
 gr-filter/include/gnuradio/filter/fir_filter.h        | 4 ++--
 gr-filter/python/filter/bindings/fir_filter_python.cc | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/gr-filter/include/gnuradio/filter/fir_filter.h b/gr-filter/include/gnuradio/filter/fir_filter.h
index 3fbd98946..1418fd67c 100644
--- a/gr-filter/include/gnuradio/filter/fir_filter.h
+++ b/gr-filter/include/gnuradio/filter/fir_filter.h
@@ -33,9 +33,9 @@ public:
     // This prevents accidentally doing needless copies, not just of fir_filter,
     // but every block that contains one.
     fir_filter(const fir_filter&) = delete;
-    fir_filter(fir_filter&&) = default;
+    fir_filter(fir_filter&&) noexcept = default;
     fir_filter& operator=(const fir_filter&) = delete;
-    fir_filter& operator=(fir_filter&&) = default;
+    fir_filter& operator=(fir_filter&&) noexcept = default;
 
     void set_taps(const std::vector<TAP_T>& taps);
     void update_tap(TAP_T t, unsigned int index);
diff --git a/gr-filter/python/filter/bindings/fir_filter_python.cc b/gr-filter/python/filter/bindings/fir_filter_python.cc
index b320b1b7a..7fca2d41b 100644
--- a/gr-filter/python/filter/bindings/fir_filter_python.cc
+++ b/gr-filter/python/filter/bindings/fir_filter_python.cc
@@ -14,7 +14,7 @@
 /* BINDTOOL_GEN_AUTOMATIC(0)                                                       */
 /* BINDTOOL_USE_PYGCCXML(0)                                                        */
 /* BINDTOOL_HEADER_FILE(fir_filter.h)                                        */
-/* BINDTOOL_HEADER_FILE_HASH(48049333c1cea0b522a397159553f186)                     */
+/* BINDTOOL_HEADER_FILE_HASH(34aa8fbb799dad34346a83097efa6819)                     */
 /***********************************************************************************/
 
 #include <pybind11/complex.h>
-- 
2.20.1

